### ISSUE TITLE
Missing Value box in Clients

### ISSUE THREAD

**Issue by NYU-AMS** [codetodesign/NYUAMS#685]
Can we add a new value box in clients called "Colleges".   As a note Colleges have departments.

Here is the full list of Colleges:

College of Arts & Science
College of Dentistry
Gallatin School of Individualized Study
Leonard N. Stern School of Business
Liberal Studies
Rory Meyers College of Nursing
Steinhardt School of Culture, Education, and Human Development
Silver School of Social Work
School of Professional Studies
Tandon School of Engineering
Tisch School of the Arts

![colleges drop down](https://cloud.githubusercontent.com/assets/5067173/22747639/a104c7b2-edf5-11e6-97b5-3091a89a8e41.jpg)
![college missing](https://cloud.githubusercontent.com/assets/5067173/22747650/a837539c-edf5-11e6-805d-7bb9e48dfab2.jpg)

**Comment by barryross** [comment]
@leobardoramon Do we want to just hard-wire the values or is this opening up a whole:

- add/edit/remove Colleges from the admin section
- store in the database

etc.

**Comment by buck0land** [comment]
This should not be hardwired. We will need to be able to edit from time to time

-Andrew

**Comment by leobardoramon** [comment]
@barryross , this feature will affect:
 - Clients/New Client WIZ
 - Clients/Update Panel

We will also need a new sub-tab in the page Admin:
 - Admin/clients
   This page will look very similar to Admin/accounts
   There will be a table very similar to that of Admin/wide/"Paper Types"

See screenshots below for (1) Admin/wide/Paper Types table and (2) Orders/"Part Calculator"/"Wide Format"

<img width="1632" alt="screen shot 2017-03-21 at 1 41 22 pm" src="https://cloud.githubusercontent.com/assets/2165796/24162214/249f34e2-0e3d-11e7-95aa-9ae1a2e30985.png">
<img width="680" alt="screen shot 2017-03-21 at 1 42 41 pm" src="https://cloud.githubusercontent.com/assets/2165796/24162213/249cbbb8-0e3d-11e7-8a03-aa76a01c6282.png">

**Comment by barryross** [comment]
@leobardoramon   How does (2) Orders/"Part Calculator"/"Wide Format"  fit into this issue?  Is that just there because that's how the list of colleges should appear in the `clients/new client` & `client/update` panels?

**Comment by leobardoramon** [comment]
yes, @barryross , "Wide Format" and "Part Calculator" are just an example
of what we'd like the UI to resemble

**Comment by barryross** [comment]
@leobardoramon gotcha, thank you

**Comment by barryross** [comment]
To clarify the flexibility needed...do you need to be able to:

a)  add/edit/remove the names of the colleges available in the drop down? 
b) modify which college a client is associated with? 
c) both

**Comment by leobardoramon** [comment]
c) both

**Comment by barryross** [comment]
I notice there is an "add new paper type" but it isn't highlighted in the image above.  I assume the answer above is "c" ?

**Comment by barryross** [comment]
@leobardoramon great, thank you

**Comment by barryross** [comment]
@leobardoramon `admin->clients` is the tab in which the admin would add/edit/remove college names to the list, correct?  And the association off college to client would only take place in a) the `new client` panel or `update client` panel, correct?

**Comment by leobardoramon** [comment]
@barryross that is correct

**Comment by barryross** [comment]
@leobardoramon thanks for confirmation!

**Comment by barryross** [comment]
@leobardoramon Is what I'm picturing here correct? 

<img width="1382" alt="screen shot 2017-12-28 at 4 44 07 pm" src="https://user-images.githubusercontent.com/2281667/34426783-f9c9f802-ebee-11e7-85c8-79520600ad4d.png">

**Comment by buck0land** [comment]
Yes it looks good to me.  Leo is this what your thinking?

**Comment by barryross** [comment]
@buck0land thanks, Andrew!

**Comment by leobardoramon** [comment]
Yes @barryross , this is great .  Thank you !

**Comment by barryross** [comment]
@leobardoramon awesome, glad to hear! You're welcome!

**Comment by leobardoramon** [comment]
@barryross ,
1. please add a dropdown ( similar to Admin/rapid ) page for (a) clients (b) colleges
2. you can model the colleges table the same way as the Admin/FDM/"Available Tips","Available Trays" tables ?

@buck0land ,
1. would you like Barry to remove any of the columns that you may not need for Admin/clients table since it's a mirror of the Clients tab ( for faster loading , etc ) ?
2. something to discuss in the future would be add/edit/delete ( would you like us to change accessibility for non-admin users , since the table now exists in the Admin tab )

**Comment by NYU-AMS** [comment]
You are good to go for removing not needed columns.  do not change the access levels this should be admin only

-Andrew

**Comment by leobardoramon** [comment]
@barryross , go ahead and remove Projects column from Admin/clients table

**Comment by barryross** [comment]
@leobardoramon  I would like to try using React as something different on the admin/clients tab: https://reactjs.org/  

What do you think?

**Comment by leobardoramon** [comment]
@barryross , sounds good to me .

@buck0land , what do you think about this ?  Please lmk if you'd like to meet about it .

**Comment by buck0land** [comment]
Go for it. Is sound good.

**Comment by barryross** [comment]
@leobardoramon @buck0land great, thanks guys!

**Comment by barryross** [comment]
@leobardoramon Hey...I have nearly completed that and it is ready for testing on `contractor_barry`. 

There is one thing that is not sitting right with me and that is how long it currently takes to load the "admin" tab.  While React is making navigation of the admin -> clients tab very quick, I am currently asking the database for all the clients and all of their projects when we click on the "admin" tab.  

We have a few options to address this right now: 

a) Eliminate the projects in the admin-> clients list, but keep them in "Clients" tab
b) Only retrieve the main client list the first time a user navigates to the "admin" tab in a given session
c) Spend some time optimizing querying the database based off of pagination settings.  For example, if we are only looking at the first 25 records, let's only ask the db for these etc. 

Notes:
"a" would be easy to implement and should speed up load time considerable
"b" this would have a long load time the first time, but should speed subsequent navigations to "admin" up
"c" this should end up being pretty user-friendly, but will require some more in-depth programming and take some time

@leobardoramon thoughts?

**Comment by leobardoramon** [comment]
@barryross , can you please look into how much extra work it would be for "c" . 

@buck0land has already given the green light for "a" , so we always have the option to fall back on that

**Comment by barryross** [comment]
@leobardoramon glad we have the green light for "a".  I think it'd probably take 5-10 hours with "c" and the sorting capabilities would be that of what is currently on the "clients" page.  I'm comfortable giving this a shot within the scope of the current issue so we can see what we like best...sound good?  

I would be inclined to try using this other table component with it: 
 https://ant.design/components/table/#components-table-demo-ajax

**Comment by leobardoramon** [comment]
@barryross , yes , do this

**Comment by barryross** [comment]
@leobardoramon Hey Leo...I didn't end up using the "ant design" one, but I did use the remove/ajax technique.  Please check it out `contractor_barry` !

**Comment by leobardoramon** [comment]
Hey @barryross  , we found a couple of things : 

1 . issues with the LDAP component in
 - Project/New Project WIZ/New Client
 - Clients/New Client PANEL
 - Admin/Client

<img width="581" alt="screen shot 2018-02-13 at 2 02 43 pm" src="https://user-images.githubusercontent.com/2165796/36168675-e23dc45e-10c7-11e8-91c6-ea705e21ce96.png">


2 . issue when switching  dropdown between college and clients and adding a new college , table says "no data to display"

See Video Here : https://drive.google.com/open?id=1KMXWWoYJOFd6r6ZuitcjVEU19QiNES-D
